cmd/virt-apiservice-register/README.md

```diff
@@ -0,0 +1,200 @@
+-service-name=myservice \
+-target-selector=app=myapiservice
+```
+
+It is important to make sure that the target, the pod where the actual
+API service is running, has labels that matches the selector given in the
+'-target-selector' option.
+
+## How to use it
+
+The tool is intended for use in an initialization container inside the
+pod of the API server itself, so that it is always executed before the
+API server container. The secret should be created before running the
+tool. The API server container should mount the secret, and use the
+files inside to configure the TLS options. For example:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+---
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: myserver
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: myserver
+    spec:
+      serviceAccountName: myaccount
+      volumes:
+      - name: pkifiles
+        secret:
+          secretName: mysecret
+      initContainers:
+      - name: register
+        image: virt-apiservice-register:latest
+        imagePullPolicy: IfNotPresent
+        args:
+        - -logtostderr
+        - -v=0
+        - -api-group=myapi.io
+        - -api-version=v1alpha1
+        - -secret-name=mysecret
+        - -service-name=myservice
+        - -target-selector=app=myserver
+      containers:
+      - name: server
+        image: myimage:latest
+        imagePullPolicy: IfNotPresent
+        args:
+        - --etcd-servers=http://localhost:2379
+        - --tls-ca-file=/mypkifiles/ca.crt
+        - --tls-cert-file=/mypkifiles/tls.crt
+        - --tls-private-key-file=/mypkifiles/tls.key
+        volumeMounts:
+        - name: pkifiles
+          mountPath: "/mypkifiles"
+          readOnly: true
+```
+
+The tool doesn't automatically approve the requested certificate, it
+will just wait for the administrator (or some other tool) to approve or
+deny it. The details will be shown in the log:
+
+```
+I: Found CA
+-----BEGIN CERTIFICATE-----
+MIICyDCCAbCgAwIBAgIBADANBgkqhkiG9w0BAQsFADAVMRMwEQYDVQQDEwprdWJl
+6UO/0vLzTdESJUz/a0RMITXICx0dt7vbPZMjOV6cS5sW4EhfA+DnhS2ZeFw=
+-----END CERTIFICATE-----
+I: Secret 'mysecret' already exists
+I: The CN for the certificate is 'myservice.default.svc.cluster.local'
+I: Additional host name for the certificate is 'myservice.default.svc.cluster.local'
+I: Additional host name for the certificate is 'myservice.default.svc'
+I: Creating CSR 'myservice.default.svc.cluster.local' for CN 'myservice.default.svc.cluster.local'
+I: The CSR has been created
+-----BEGIN CERTIFICATE REQUEST-----
+MIICyzCCAbMCAQAwLjEsMCoGA1UEAxMjbXlzZXJ2aWNlLmRlZmF1bHQuc3ZjLmNs
+35E81nwF+X7ctMUh6F29Gb2oYulCFir3W3gpeTchxmNyqv7Ebpm9aeiBeIbICzk=
+-----END CERTIFICATE REQUEST-----
+I: Waiting for approval of the CSR
+```
+
+At this point the administrator can approve the certificate with the
+following command:
+
+```shell
+$ kubctl certificate approve myservice.default.svc.cluster.local
```

Comment:
given the right permissions, could this tool automatically approve the cert?  I'd like that to be a feature of the tool if there's no technical limitations.

Reply:
Yes, we can add a `-auto-approve' (or similar) option to indicate that the tool should try to approve itself the certificate.

Reply:
nice! :+1: